recipes/folly/all/conanfile.py

```diff
@@ -19,96 +18,90 @@ class FollyConan(ConanFile):
     homepage = "https://github.com/facebook/folly"
     license = "Apache-2.0"
 
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
-        "use_sse4_2" : [True, False],
+        "use_sse4_2": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
-        "use_sse4_2" : False
+        "use_sse4_2": False
     }
 
-    generators = "cmake", "cmake_find_package"
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return "17" if Version(self.version) >= "2022.01.31.00" else "14"
 
     @property
-    def _minimum_cpp_standard(self):
-        return 17 if Version(self.version) >= "2022.01.31.00" else 14
-
-    @property
-    def _minimum_compilers_version(self):
+    def _compilers_minimum_version(self):
         return {
             "Visual Studio": "15",
             "gcc": "5",
             "clang": "6",
             "apple-clang": "8",
-        } if self._minimum_cpp_standard == 14 else {
+        } if self._min_cppstd == "14" else {
             "gcc": "7",
             "Visual Studio": "16",
             "clang": "6",
             "apple-clang": "10",
         }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
-
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-
         if str(self.settings.arch) not in ['x86', 'x86_64']:
             del self.options.use_sse4_2
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.78.0")
+        self.requires("boost/1.78.0", transitive_headers=True)
         self.requires("bzip2/1.0.8")
-        self.requires("double-conversion/3.2.0")
-        self.requires("gflags/2.2.2")
-        self.requires("glog/0.4.0")
-        self.requires("libevent/2.1.12")
-        self.requires("openssl/1.1.1q")
+        self.requires("double-conversion/3.2.1", transitive_headers=True)
+        self.requires("gflags/2.2.2", transitive_headers=True)
+        self.requires("glog/0.4.0", transitive_headers=True)
+        self.requires("libevent/2.1.12", transitive_headers=True)
+        self.requires("openssl/1.1.1t", transitive_headers=True)
         self.requires("lz4/1.9.3")
         self.requires("snappy/1.1.9")
-        self.requires("zlib/1.2.12")
-        self.requires("zstd/1.5.2")
+        self.requires("zlib/1.2.13", transitive_headers=True)
+        self.requires("zstd/1.5.2", transitive_headers=True)
```

Comment:
Looking at the error for v1 it's also missing transitive libs in a few places. Please add a comment if you add it so we know :)

Reply:
Ok. Still not 100% sure how this transitive model works with conan 2.0. I will make the corrections and add comments. Just heads up if you see something missing.

Reply:
@prince-chrismc would it be better to not specify transitive attributtes here? (and get the default behavior?)